### ISSUE TITLE
Allow additional information in metadata.yaml

### ISSUE THREAD

**Issue by slochower** [greenelab/manubot-rootstock#129]
Based on some of the points already discussed on deep review and https://github.com/greenelab/meta-review/pull/75#discussion_r202456488, I think adding a few additional variables to the metadata would help Manubot be a little more flexible. Some ideas of what we might want to allow:

- Address information (in addition to affiliation)
- Corresponding author status
- First/co-first author status
- Specification of author symbol

For the last three, I think we could implement sensible defaults in the jinja template to use if not specified. For example, corresponding author status may be set to "no" unless it is explicitly set to "yes."

**Comment by agitter** [comment]
This relates to some early discussion in #7 and #10.  The debate before was whether some annotations are common enough to warrant special support or whether the authors should be add custom annotations with symbols.  Co-first and corresponding annotations seem fairly common to me.  Other annotations (current address, co-second author, deceased, etc.) are infrequent, but may be a good reason for general symbol support.

Now that we've moved to yaml instead of tsv for author metadata, I'm more supportive of including optional attributes like corresponding.  As @slochower says, they could be ignored by default.

**Comment by agitter** [comment]
This could also be a good time to re-assess whether we want to make the `affiliations` field mandatory.  That would allow us to automatically check whether the affiliations were provided correctly.

**Comment by slochower** [comment]
> This could also be a good time to re-assess whether we want to make the affiliations field mandatory.

And consider having an option for condensing multiple affiliations into one symbol? 

My general thinking is that as many things that journals might "require" should be configured via flags like `condense_affiliation: yes` in the `yaml` file so that changing a manuscript to resubmit to another journal would only require changing things in `content/` and not editing the python. Do you agree?

**Comment by agitter** [comment]
I agree that it would be better to make manuscripts configurable with the yaml instead of editing the Manubot source.  Some of these changes would require Manubot changes.  Others, like symbols, would be easier and could be demonstrated with updates to Manubot rootstock alone.

**Comment by agitter** [comment]
> To accommodate corresponding authors & other symbols as well as affiliation addresses, we could think about making this more accessible via manubot-rootstock.

Adding an example of frontmatter symbols to manubot-rootstock would be a small change that doesn't require additional maintenance.  It also provides a lot of flexibility for whatever field-specific annotations an author finds important.

**Comment by slochower** [comment]
> @slochower, you seem to be interested in making the Manubot frontmatter match the style of a specific journal.

I'm actually not trying to make the frontmatter match a *specific* journal. On the contrary: I want Manubot to be able to match a bunch of "common" journal requirements (of course we can debate what is "common," as @agitter mentioned -- I'm not talking about dozens of options here, just a few common ones). I've lost a lot of time reformatting documents for different journals. If I have to muck around in the code when I need to submit to a different journal, I've just replaced time messing with Word documents with time spent writing and testing the Python. Does that make sense?

**Comment by dhimmel** [comment]
Noting that https://github.com/manubot/rootstock/commit/9504aa7751e9d5ca0e7df896bf3aeeb73d2fb262 adds support for `author.corresponding` in `metadata.yaml`.
